prov/psm/src/psmx_init.c

```diff
@@ -194,6 +193,13 @@ static int psmx_getinfo(uint32_t version, const char *node, const char *service,
 			goto err_out;
 		}
 
+		if (hints->fabric_attr && hints->fabric_attr->prov_name &&
+		    strncmp(hints->fabric_attr->prov_name, "psm", 3)) {
+			psmx_debug("%s: hints->fabric_prov_name=%s, supported=psm\n",
+					__func__, hints->fabric_attr->prov_name);
+			goto err_out;
+		}
+
```

Comment:
Let's move the provider name check into fabric.c.  There's already a check there for fi_fabric().  We can also add a check to fi_getinfo(), to avoid duplicating it in every provider.